nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/eventhub/PutAzureEventHub.java

```diff
@@ -161,25 +170,38 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         }
 
         final StopWatch stopWatch = new StopWatch(true);
-            final byte[] buffer = new byte[(int) flowFile.getSize()];
-            session.read(flowFile, in -> StreamUtils.fillBuffer(in, buffer));
+        final Map<String, String> attributes = new HashMap<>();
+        final byte[] buffer = new byte[(int) flowFile.getSize()];
+        session.read(flowFile, in -> StreamUtils.fillBuffer(in, buffer));
+
+        final PropertyValue idPropertyValue = context.getProperty(PARTITION_KEY);
+
+        final String partitionId = (null == idPropertyValue || !idPropertyValue.isSet()) ?
+                UUID.randomUUID().toString() : idPropertyValue.getValue();
```

Comment:
Do we have to generate UUID here? Or should we let EventHub library handle that? How does the library determine partition if we doesn't pass any partition key to sendSync?